Conch/lib/Conch/Model/SessionToken.pm

```diff
@@ -0,0 +1,120 @@
+=pod
+
+=head1 NAME
+
+Conch::Model::SessionToken
+
+=head1 METHODS
+
+=cut
+
+package Conch::Model::SessionToken;
+use Mojo::Base -base, -signatures;
+
+use Conch::UUID qw(is_uuid);
+use Session::Token;
+
+has [
+	qw(
+		email
+		id
+		name
+		password_hash
+		)
+];
+
+=head2 create
+
+Create and record a session token
+
+=cut
+
+sub create ( $class, $user_id, $expires ) {
+	my $token = Session::Token->new->get;
+
+	Conch::Pg->new->db->query(
+		q{
+			insert into user_session_token
+				(user_id, token_hash, expires)
+			values( ?, digest(?, 'sha256'), to_timestamp(?)::timestamptz)
+		},
+		$user_id,
+		$token,
+		$expires,
+	);
+	return $token;
+}
+
+=head2 check_token
+
+Check if a session token is valid for the user ID. Returns 1 if valid, 0 otherwise
+
+=cut
+
+sub check_token ( $class, $user_id, $token ) {
+
+	Conch::Pg->new->db->query( 'select delete_expired_tokens()' );
```

Comment:
Can we make this a method or something that has a descriptive name? While the sql is clear, I had to go to the sql to verify that this was acting on the daatbase and wasn't just a select that we were throwing away.

Reply:
We've been talking about stored procedures, and I saw this as a way to lightly try them out (originally, I was going to use this function in a trigger to force ensure you could never query for expired tokens, but then I realized that automatically disappearing data wasn't the best idea...).

This disconnect between the code and the query is the biggest reason I am hesitant to use stored procedures. It certainly has its use in more complex queries (like the `workspace_devices` procedure I made yesterday), but is probably not appropriate here. I'd prefer to go ahead and replace the invocation with the actual queries.

Reply:
You're reusing the query so either way it should be abstracted a bit. I have no problem with the stored procedure itself but it just needs to be a bit more clear in the code, via either a good method name or docs or both.